application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,265 @@
+            @NotNull SelectionMenu.Builder menu, @NotNull String placeHolder,
+            @Nullable Integer minValues) {
+
+        Role highestBotRole =
+                Objects.requireNonNull(event.getGuild()).getSelfMember().getRoles().get(1);
+        List<Role> guildRoles = Objects.requireNonNull(event.getGuild()).getRoles();
+
+        List<Role> roles = new ArrayList<>(
+                guildRoles.subList(guildRoles.indexOf(highestBotRole) + 1, guildRoles.size()));
+
+        if (minValues != null) {
+            menu.setMinValues(minValues);
+        }
+
+        menu.setPlaceholder(placeHolder).setMaxValues(roles.size());
+
+        for (Role role : roles) {
+            if (role.getName().equals("@everyone"))
+                continue;
+            menu.addOption(role.getName(), role.getId());
+        }
+    }
+
+    /**
+     * Creates an embedded message to send with the selection menu
+     *
+     * <p>
+     * </p>
+     *
+     * @param title for the embedded message. nullable {@link String}
+     * @param description for the embedded message. nullable {@link String}
+     * @return the formatted embed {@link MessageEmbed}
+     */
+    private static @NotNull MessageEmbed makeEmbed(@Nullable String title,
+            @Nullable String description) {
+
+        if (title == null) {
+            title = "Select your roles:";
+        }
+
+        EmbedBuilder embedBuilder = new EmbedBuilder();
+        embedBuilder.setTitle(title);
+
+        if (description != null) {
+            embedBuilder.setDescription(description);
+        }
+
+        embedBuilder.setColor(new Color(24, 221, 136));
+
+        return embedBuilder.build();
+    }
+
+    @Override
+    public void onSelectionMenu(@NotNull SelectionMenuEvent event, @NotNull List<String> args) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+
+        // Get the roles the bot can interact with
+        List<Role> guildRoles = Objects.requireNonNull(event.getGuild()).getRoles();
+        List<String> roleIds = guildRoles.stream().map(Role::getId).collect(Collectors.toList());
+
+        List<Role> selectedRoles = new ArrayList<>();
+        for (SelectOption role : Objects.requireNonNull(event.getSelectedOptions())) {
+            if (roleIds.contains(role.getValue())) {
+                selectedRoles.add(guildRoles.get(roleIds.indexOf(role.getValue())));
+            }
+        }
+
+        // True if the event option was 'choose'
+        if (event.getMessage().isEphemeral()) {
+
+            SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+            menu.setPlaceholder("Select your roles").setMaxValues(selectedRoles.size());
+
+            for (SelectOption roleOption : Objects.requireNonNull(event.getSelectedOptions())) {
+                Role role = guildRoles.get(roleIds.indexOf(roleOption.getValue()));
+                menu.addOption(role.getName(), role.getId());
+            }
```

Comment:
2 things.

I doubt you need the Objects#requireNonNull, it'll throw an NPE either way to my knowledge.

And the SelectOption is named "role" here, which is really misinforming the user since it's not a Role.
It's a SelectOption, thoughts on renaming it to a.e selectedOption?